### PR TITLE
RECIPE-40

### PR DESCRIPTION
**Pull request by PatsiChang**
Added custom annotation and interceptor to validate if token is included
Link: https://smart-home-cloud.atlassian.net/browse/RECIPE-40

### CODE REVIEW

File: src/main/java/com/patsi/controller/SocialMediaUserController.java [hunk 1]
```diff
@@ -37,7 +41,10 @@ public SocialMediaUser createSocialMediaAccount(@RequestBody SocialMediaUser use
     }
 
     @PostMapping("/getUserByToken")
```
Comment (ChakShingLo) [inline src/main/java/com/patsi/controller/SocialMediaUserController.java]: This should be Get instead of Post

File: src/main/java/com/patsi/controller/SocialMediaUserController.java [hunk 2]
```diff
@@ -37,7 +41,10 @@ public SocialMediaUser createSocialMediaAccount(@RequestBody SocialMediaUser use
...
-    public SocialMediaUser getUserById(@RequestHeader("Authorization") String token) {
+    @RequireLoginSession
+    public SocialMediaUser getUserById() {
+        String token = (String) request.getAttribute("token");
```
Comment (ChakShingLo) [inline src/main/java/com/patsi/controller/SocialMediaUserController.java]: Extract `request.getAttribute("token");` to a service. You may add it to the existing UserProfileService, or create a new service such as UserSessionService.

File: src/main/java/com/patsi/utils/FileHelper.java [hunk 3]
```diff
@@ -8,12 +8,13 @@ public class FileHelper {
     public static void newFile(String pathname, String profilePictureID, byte[] image)
         throws IOException {
         File f = new File(pathname + profilePictureID + ".jpg");
+        if(f.exists()) {
```
Comment (ChakShingLo) [inline src/main/java/com/patsi/utils/FileHelper.java]: why this changes is required for this PR?

File: src/main/java/com/patsi/controller/SocialMediaUserController.java (hunk 1 above)
Comment (PatsiChang) [reply src/main/java/com/patsi/controller/SocialMediaUserController.java]: This fixed at ticket: RECIPE-45

File: src/main/java/com/patsi/service/UserProfileService.java [hunk 4]
```diff
@@ -10,7 +12,12 @@
 @Service
 public class UserProfileService {
 
-    public String getUidFromToken(String token) {
+    @Autowired
+    private HttpServletRequest request;
+
+    public String getUidFromToken() {
+        String token = (String) request.getAttribute("token");
+        System.out.println("token in getUidFromToken"+ token);
```
Comment (ChakShingLo) [inline src/main/java/com/patsi/service/UserProfileService.java]: use sl4j logger

**Review by ChakShingLo — APPROVED** [review]
lgtm
